npm/webpack-dev-server/src/aut-runner.ts

```diff
@@ -26,9 +26,11 @@ export function init (importPromises, parent = (window.opener || window.parent))
   Cypress.onSpecWindow(window, importPromises)
   Cypress.action('app:window:before:load', window)
 
-  beforeEach(() => {
-    // This really have no sense for @cypress/react and @cypress/vue
-    // It is anyway required to mount a new
+  // Before all tests we are mounting the root element, **not beforeEach**
+  // Cleaning up platform between tests is the responsibility of the specific adapter
+  // because unmounting react/vue component should be done using specific framework API
+  // (for devtools and to get rid of global event listeners from previous tests.)
```

Comment:
The comment is a little confusing to me - Just for my understanding, unmounting the component won't actually get rid of the global listener (__REACT_DEVTOOLS_GLOBAL_HOOK__) though, right? It will just make sure the component is no longer present in the devtools.

Reply:
I mean if you are doing `document.addEventListener` inside the `useEffect` and unsubscribing in return

Reply:
Right, sure. Makes sense. One thing we might need to be careful is if the devtools do `window.addEventListener('postMessage'). Vue DevTools do this - we need to clean it up because `window.addEventListener` will not replace the previous one but duplicate it.